extensions/markdown-language-features/src/features/preview.ts

```diff
@@ -434,14 +434,14 @@ class MarkdownPreview extends Disposable implements WebviewResourceProvider {
 	private async onDidClickPreviewLink(href: string) {
 		let [hrefPath, fragment] = decodeURIComponent(href).split('#');
 
-		// We perviously already resolve absolute paths.
-		// Now make sure we handle relative file paths
 		if (hrefPath[0] !== '/') {
-			// Fix #93691, use this.resource.fsPath instead of this.resource.path
-			hrefPath = path.join(path.dirname(this.resource.fsPath), hrefPath);
+			// We perviously already resolve absolute paths.
+			// Now make sure we handle relative file paths
+			const dirname = path.dirname(this.resource.path);
+			hrefPath = vscode.Uri.file(path.join(dirname, hrefPath)).path;
```

Comment:
Instead of using `path.join` and `Uri.file`, can you switch to use `Uri.joinPaths` instead? 

Also since this changes to use `fsPath` instead of `path`, please test to make sure this works properly for UNC paths and networks share

Reply:
Thanks.

- `Uri.joinPaths` works

I'm using `fsPath` caused the original issue so I use `path` instead. Will check with UNC paths and network share 🤞

Reply:
Update: I just checked network share and both this PR, existing VSCode and VSCode Insider can't open the link. Will take me a while to figure out this one

Reply:
@mjbvz I'm not familiar with network share and UNC paths. Happy to let someone take over/close the PR.